src/Diagnostics.DataProviders/DataProviderConfigurations/GenericMdmDataProviderConfiguration.cs

```diff
@@ -32,4 +35,65 @@ public void PostInitialize()
         {
         }
     }
+
+    public class GenericMdmDataProviderConfigurationWapper: IDataProviderConfiguration, IMdmDataProviderConfiguration
+    {
+        /// <summary>
+        /// Gets or sets the base endpoint.
+        /// </summary>
+        public string Endpoint { get; set; }
+
+        /// <summary>
+        /// Gets or sets the certificate thumbprint.
+        /// </summary>
+        public string CertificateThumbprint { get; set; }
+
+        /// <summary>
+        /// Gets or sets monitoring account.
+        /// </summary>
+        public string MonitoringAccount { get; set; }
+
+        /// <summary>
+        /// Post initialize.
+        /// </summary>
+        public void PostInitialize()
+        {
+        }
+
+        public GenericMdmDataProviderConfigurationWapper(GenericMdmDataProviderConfiguration config)
+        {
+            if (config == null)
+            {
+                throw new ArgumentNullException("Configuration is null.");
+            }
+            if (config.MonitoringAccount != null && config.MonitoringAccount.StartsWith("Mock"))
+            {
+                CertificateThumbprint = "BAD0BAD0BAD0BAD0BAD0BAD0BAD0BAD0BAD0BAD0";
+            }
+            else
+            {
+                X509Store store = new X509Store(StoreName.My, StoreLocation.LocalMachine);
+                store.Open(OpenFlags.ReadOnly);
+                X509Store userStore = new X509Store(StoreName.My, StoreLocation.CurrentUser);
+                userStore.Open(OpenFlags.ReadOnly);
+
+                var certificateCollection = store.Certificates;
+                certificateCollection.AddRange(userStore.Certificates);
+                var certificates = certificateCollection.Find(X509FindType.FindBySubjectName, config.CertificateName, true);
+
+                if (certificates.Count == 0)
+                {
+                    throw new SystemException("Certificate is not found.");
+                }
+
+                CertificateThumbprint = certificates[0].Thumbprint;
+
+                store.Dispose();

```

Comment:
>store.Dispose(); [](start = 16, length = 16)

use 'using (X509Store store = new X509Store(StoreName.My, StoreLocation.LocalMachine)) {...}', this will call dispose() even when an exception is thrown